### PR TITLE
Account_Update

### PR DESCRIPTION
**Pull request by VStetsyshyn**

### CODE REVIEW

File: app/controllers/accounts_controller.rb [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
-
+  before_action :check_admin_access, only: %i[index]
   def index
     @accounts = Account.all
   end
 
   def show; end
 
   def new
+    redirect_to account_path(current_user.account) unless current_user.account.nil?
```
Comment (BohdanChaban) [inline app/controllers/accounts_controller.rb]: - Here should be used `if`, not unless
- Maybe `current_account_path`?

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -4,9 +4,9 @@
       <%= link_to 'News', root_path, class: 'btn btn-warning' %>
       <%= link_to 'About Us', abouts_path, class: 'btn btn-warning' %>
       <%= link_to 'Contacts', contacts_path, class: 'btn btn-warning' %>
-      <% if user_signed_in? %>
+      <% if user_signed_in? && current_user.account %>
```
Comment (BohdanChaban) [inline app/views/layouts/_header.html.erb]: How make LogOut when user_signed_in but without account?

Comment (VStetsyshyn) [reply app/views/layouts/_header.html.erb]: How better to do, like now I update or like this using account_helper to avoiding overused if statements in html?
```module AccountsHelper
    def link(account)
    account ? account_path(account) : ''
  end
end
```
*html*
   <%= link_to 'Account', link(current_user.account), class: 'btn btn-warning right' %>

File: app/controllers/accounts_controller.rb (hunk 1 above)
Comment (VStetsyshyn) [reply app/controllers/accounts_controller.rb]: current_account_path not working in header_html

File: app/assets/stylesheets/accounts.scss [hunk 3]
```diff
@@ -1,3 +1,13 @@
 // Place all the styles related to the Accounts controller here.
 // They will automatically be included in application.css.
 // You can use Sass (SCSS) here: http://sass-lang.com/
+.btn-primary a {
+	color:yellow;
+	background-color:black;
+}
+.father {
```
Comment (BohdanChaban) [inline app/assets/stylesheets/accounts.scss]: better to use name `parent` for selector

File: app/views/accounts/edit.html.erb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <h1>Editing Account</h1>
 
 <%= render 'form', account: @account %>
+  <%= link_to '<button type="button" class="btn btn-light">Show</button>'.html_safe, @account %>
 
-<%= link_to 'Show', @account %> |
-<%= link_to 'Back', accounts_path %>
+<%= link_to '<button type="button" class="btn btn-light">Back</button>'.html_safe, account_path(current_user.account) %>
```
Comment (BohdanChaban) [inline app/views/accounts/edit.html.erb]: missed end line

File: app/views/contacts/index.html.erb [hunk 5]
```diff
@@ -1,32 +1,30 @@
...
-        <tr><div class = "table_name"><b><%= contact.name %></b></div>
-       </tr>
-
-      <%=show_non_blank_field "Address", contact.address %>
-       
+      <tbody>
+        <tr>
+          <div class = "table_name"><b><%= contact.name %></b></div>
+        </tr>
+        <%=show_non_blank_field "Address", contact.address %>
```
Comment (BohdanChaban) [inline app/views/contacts/index.html.erb]: missed indent `before show_non_blank_field`

File: app/views/layouts/_header.html.erb [hunk 6]
```diff
@@ -4,9 +4,13 @@
       <%= link_to 'News', root_path, class: 'btn btn-warning' %>
       <%= link_to 'About Us', abouts_path, class: 'btn btn-warning' %>
       <%= link_to 'Contacts', contacts_path, class: 'btn btn-warning' %>
-      <% if user_signed_in? %>
+      <% if user_signed_in?  %>
```
Comment (BohdanChaban) [inline app/views/layouts/_header.html.erb]: unnecessary indent after `user_signed_in?`

File: app/views/layouts/_header.html.erb [hunk 7]
```diff
@@ -4,9 +4,13 @@
...
         <%= link_to 'Log Out', destroy_user_session_path, method: :delete, class: 'btn btn-warning right' %>
-        <%= link_to 'Account', '', class: 'btn btn-warning right' %>
+        <% if current_user.account %>
+          <%= link_to 'Account', account_path(current_user.account), class: 'btn btn-warning right' %>
```
Comment (BohdanChaban) [inline app/views/layouts/_header.html.erb]: better use here ternary operator

File: app/helpers/accounts_helper.rb [hunk 8]
```diff
@@ -1,2 +1,5 @@
 module AccountsHelper
+  # def link(account)
+  #   account ? account_path(account) : ''
+  # end
```
Comment (BohdanChaban) [inline app/helpers/accounts_helper.rb]: remove it pls

File: app/views/layouts/_header.html.erb (hunk 6 above)
Comment (BohdanChaban) [reply app/views/layouts/_header.html.erb]: Not fixed

File: app/views/layouts/_header.html.erb [hunk 9]
```diff
@@ -4,9 +4,11 @@
       <%= link_to 'News', root_path, class: 'btn btn-warning' %>
       <%= link_to 'About Us', abouts_path, class: 'btn btn-warning' %>
       <%= link_to 'Contacts', contacts_path, class: 'btn btn-warning' %>
-      <% if user_signed_in? %>
+      <% if user_signed_in?  %>
         <%= link_to 'Log Out', destroy_user_session_path, method: :delete, class: 'btn btn-warning right' %>
-        <%= link_to 'Account', '', class: 'btn btn-warning right' %>
+        <%= current_user.account ? 
+        (link_to 'Account', account_path(current_user.account), class: 'btn btn-warning right') : 
+        (link_to 'Account', new_account_path, class: 'btn btn-warning right') %>
```
Comment (BohdanChaban) [inline app/views/layouts/_header.html.erb]: you should use ternary operator inside link_to. Use only one link_to

**Pull request merged**